### PR TITLE
feature(storage): support gcs blob storage

### LINKED ISSUE #639: Support backend storage driver: Google Cloud Storage bucket
See https://cloud.google.com/storage

### PR DESCRIPTION
**Pull request by soulseen**
fix #639 

Signed-off-by: soulseen <zhuxiaoyang1996@gmail.com>

### CODE REVIEW

File: apis/goharbor.io/v1beta1/chartmuseum_types.go [hunk 1]
```diff
@@ -158,6 +158,25 @@ type ChartMuseumChartStorageDriverSpec struct {
...
 	Azure *ChartMuseumChartStorageDriverAzureSpec `json:"azure,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	Gcs *ChartMuseumChartStorageDriverGcsSpec `json:"gcs,omitempty"`
+}
+
+type ChartMuseumChartStorageDriverGcsSpec struct {
+	// +kubebuilder:validation:Required
+	// bucket to store charts for Gcs storage
+	Bucket string `json:"bucket,omitempty"`
```
Comment (cndoit18) [inline apis/goharbor.io/v1beta1/chartmuseum_types.go:169]: remove `omitempty`

File: apis/goharbor.io/v1beta1/chartmuseum_types.go [hunk 2]
```diff
@@ -158,6 +158,25 @@ type ChartMuseumChartStorageDriverSpec struct {
...
+
+	// +kubebuilder:validation:Required
+	// The base64 encoded json file which contains the key
+	KetDataSecretRef string `json:"ketDataSecretRef,omitempty"`
```
Comment (cndoit18) [inline apis/goharbor.io/v1beta1/chartmuseum_types.go:173]: remove `omitempty`

File: apis/goharbor.io/v1beta1/harborcluster_types.go [hunk 3]
```diff
@@ -212,7 +213,7 @@ type ZlandoPostgreSQLSpec struct {
 
 type Storage struct {
 	// Kind of which storage service to be used. Only support MinIO now.
-	// +kubebuilder:validation:Enum={MinIO,S3,Swift,FileSystem,Azure}
+	// +kubebuilder:validation:Enum={MinIO,S3,Swift,FileSystem,Azure, Gcs}
```
Comment (cndoit18) [inline apis/goharbor.io/v1beta1/harborcluster_types.go:216]:
```suggestion
	// +kubebuilder:validation:Enum={MinIO,S3,Swift,FileSystem,Azure,Gcs}
```

**Review by cndoit18 — CHANGES_REQUESTED** [review]

**Review by cndoit18 — APPROVED** [review]
LGTM

**Review by bitsf — APPROVED** [review]

**Pull request merged**